scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1953 @@
+
+    This lets us plot everything from C's perspective:
+
+    >>> fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+    >>> plot_transformed_axes(ax, tf_C, name="tfC")     # C plotted in C
+    >>> plot_transformed_axes(ax, tf_C_B, name="tfCB")  # B plotted in C
+    >>> plot_transformed_axes(ax, tf_C_A, name="tfCA")  # A plotted in C
+    >>> ax.scatter(p1_C[0], p1_C[1], p1_C[2], color=colors[0])
+    >>> ax.scatter(p2_C[0], p2_C[1], p2_C[2], color=colors[1])
+    >>> ax.set_title("A, B, C frames and points with respect to C")
+    >>> ax.set_aspect("equal")
+    >>> ax.figure.set_size_inches(6, 5)
+    >>> plt.show()
+    """
+
+    cdef double[:, :, :] _matrix
+    cdef bint _single
+
+    def __init__(self, matrix, normalize=True, copy=True):
+        """Initialize from a 4x4 transformation matrix.
+
+        Rotations are not meant to be initialized directly. Please use one of
+        the `from_...` methods instead.
+
+        Parameters
+        ----------
+        matrix : array_like, shape (4, 4) or (N, 4, 4)
+            A single transformation matrix or a stack of transformation
+            matrices.
+        normalize : bool, optional
+            If True, orthonormalize the rotation matrix using singular value
+            decomposition.
+        copy : bool, optional
+            If True, copy the input matrix. If False, a reference to the input
+            matrix is used. If normalize is True, the input matrix is always
+            copied regardless of the value of copy.
+
+        Returns
+        -------
+        transformation : `RigidTransformation` instance
+        """
+        self._single = False
+        matrix = np.asarray(matrix, dtype=float)
+
+        if (matrix.ndim not in [2, 3]
+                or matrix.shape[0] == 0
+                or matrix.shape[-1] != 4
+                or matrix.shape[-2] != 4):
+            raise ValueError("Expected `matrix` to have shape (4, 4), or (N, 4, 4), "
+                             f"got {matrix.shape}.")
+
+        # If a single transformation matrix is given, convert it to a
+        # 2D 1 x 4 x 4 matrix but set self._single to True so that we can
+        # return appropriate objects in the `to_...` methods
+        if matrix.shape == (4, 4):
+            matrix = matrix[None, :, :]
+            self._single = True
+
+        if normalize or copy:
+            matrix = matrix.copy()
+
+        # Rigid transformations have the following matrix representation:
+        # [R | t]
+        # [0 | 1]
+        # where R is a 3x3 orthonormal rotation matrix and t is a 3x1 translation
+        # vector. The last row is always [0, 0, 0, 1] exactly
+
+        # Check the last row. Exact match is required, as this last row should
+        # not accumulate floating point error during composition.
+        last_row_ok = np.all(matrix[:, 3, :] == np.array([0, 0, 0, 1]), axis=1)
+        if np.any(~last_row_ok):
+            ind = np.where(~last_row_ok)[0][0]
```

Comment:
The result will be a scalar index, is it what you want?

Reply:
My thought was that a full list of the invalid indices would be too much for a debug statement.